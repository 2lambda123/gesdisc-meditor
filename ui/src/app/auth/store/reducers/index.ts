import { createSelector, createFeatureSelector } from '@ngrx/store';
import * as fromAuth from './auth';

export interface AuthState {
  status: fromAuth.State;
}


export const reducers = {
  status: fromAuth.reducer
};

export const selectAuthState = createFeatureSelector<AuthState>('auth');

export const selectAuthStatusState = createSelector(
  selectAuthState,
  (state: AuthState) => state.status
);

export const getLoggedIn = createSelector(
  selectAuthStatusState,
  fromAuth.getLoggedIn
);

export const getUser = createSelector(
  selectAuthStatusState, 
  fromAuth.getUser
);