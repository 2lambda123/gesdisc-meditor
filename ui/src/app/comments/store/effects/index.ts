import { CommentsEffects } from './comments.effects';

export const effects: any[] = [CommentsEffects];

export * from './comments.effects';