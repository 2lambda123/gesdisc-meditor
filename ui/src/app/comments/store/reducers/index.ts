import {
	createFeatureSelector,
	ActionReducerMap,
} from '@ngrx/store';
import * as fromComments from './comments.reducer';

export interface CommentsState {
	comments: fromComments.State;
}

export const reducers: ActionReducerMap<CommentsState> = {
	comments: fromComments.reducer
};

export const getCommentsState = createFeatureSelector<CommentsState>('comments');



