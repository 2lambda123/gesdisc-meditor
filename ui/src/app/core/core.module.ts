import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
