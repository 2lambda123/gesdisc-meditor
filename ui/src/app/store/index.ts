export { DocumentState } from './document/document.state';
export { ModelState } from './model/model.state';
export { NotificationState } from './notification/notification.state';
export { AuthState } from './auth/auth.state';
export { RouterState } from './router/router.state';
