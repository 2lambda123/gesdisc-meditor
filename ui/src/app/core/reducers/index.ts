import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
