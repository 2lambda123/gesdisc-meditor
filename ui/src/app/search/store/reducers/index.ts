import {
	ActionReducerMap,
	createFeatureSelector
} from '@ngrx/store';

import * as fromSearch from './results.reducer';
import * as fromRoot from '../../../store';

export interface State {
	results: fromSearch.State;
}

export const reducers: ActionReducerMap<State, any> = {
	results: fromSearch.reducer
};

export const getSearchState = createFeatureSelector<State>(
  'search'
);
