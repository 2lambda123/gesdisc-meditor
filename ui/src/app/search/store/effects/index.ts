import { ResultEffects } from './result.effects';

export const effects: any[] = [ResultEffects];

export * from './result.effects';