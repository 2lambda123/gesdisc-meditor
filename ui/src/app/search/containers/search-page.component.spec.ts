import { ComponentFixture, TestBed  } from '@angular/core/testing';
import { NgxsModule, Store, } from '@ngxs/store';
import { SearchPageComponent } from './search-page.component';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('Search Page', () => {
	let fixture: ComponentFixture<SearchPageComponent>;
	let store: Store;
	let component: SearchPageComponent;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [
				NoopAnimationsModule,
        NgxsModule.forRoot()
			],
			declarations: [
				SearchPageComponent
			],
			schemas: [ NO_ERRORS_SCHEMA ]
		});

		fixture = TestBed.createComponent(SearchPageComponent);
		component = fixture.componentInstance;
		store = TestBed.get(Store);
	});

	it('should compile', () => {
		fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
	});

});
