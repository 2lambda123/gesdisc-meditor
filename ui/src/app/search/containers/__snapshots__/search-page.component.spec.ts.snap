// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page should compile 1`] = `
<med-search-page
  filteredResults$={[Function Store]}
  models$={[Function Store]}
  results$={[Function Store]}
  rootStore={[Function Store]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  searchStore={[Function Store]}
  selectedModel$={[Function Store]}
>
  <med-search-bar
    _ngcontent-c0=""
  /><med-search-status
    _ngcontent-c0=""
  /><med-search-result-list
    _ngcontent-c0=""
  />
</med-search-page>
`;
